### PR TITLE
build(issues1033): switch log appender from DailyRollingFileAppender to RollingFileAppender

### PR DESCRIPTION
**Pull request by warr99**
* Total disk usage: 5,080MB (less than 5GB)
* Details：
image

### CODE REVIEW

**Comment by zhouxinyu** [comment]
Is there any compatible issue if we upgrade the cluster from an old version that already generated some log files? Will the old log files be deleted as expected?

File: tests/kafkatest/services/kafka/templates/log4j.properties [hunk 1]
```diff
@@ -20,86 +20,98 @@ log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
 log4j.appender.stdout.layout.ConversionPattern=[%d] %p %m (%c)%n
 
 # INFO level appenders
-log4j.appender.kafkaInfoAppender=org.apache.log4j.DailyRollingFileAppender
-log4j.appender.kafkaInfoAppender.DatePattern='.'yyyy-MM-dd-HH
+log4j.appender.kafkaInfoAppender=org.apache.log4j.RollingFileAppender
```
Comment (superhx) [inline tests/kafkatest/services/kafka/templates/log4j.properties:23]: tests log4j.properties don't need to change

**Comment by warr99** [comment]
The `RollingFileAppender` can only delete files named as `fileName.maxBackupIndex`, and it will not be able to correctly delete log files generated by older versions.

File: tests/kafkatest/services/kafka/templates/log4j.properties (hunk 1 above)
Comment (warr99) [reply tests/kafkatest/services/kafka/templates/log4j.properties:23]: Alright, I'll make a change.

**Comment by zhouxinyu** [comment]
> The `RollingFileAppender` can only delete files named as `fileName.maxBackupIndex`, and it will not be able to correctly delete log files generated by older versions.

Got it!

**Review by superhx — APPROVED** [review]

**Pull request merged**